### ISSUE TITLE
Stack overflow on setup

### ISSUE THREAD

**Issue by wookayin** [mg979/tabline.nvim#11]
When executing
```
lua require'tabline.setup'.setup()
```
the following error happens:

```
E5108: Error executing lua stack overflow
stack traceback:
        [C]: in function 'require'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:150: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:159: in function 'load_theme'
        ...local/share/nvim/lazy/tabline.nvim/lua/tabline/setup.lua:193: in function 'setup'
        [string ":lua"]:1: in main chunk
```

Indeed the logic in `M.load_theme` seems buggy, it will be an infinite recursion if the following line fails:

```
      loaded, theme = pcall(require, 'tabline.themes.' .. M.settings.theme)
```

**Comment by wookayin** [comment]
The inner stack trace is lost due as being swallowed, but the real issue is that the `tabline.highlight` module can't be imported.

```
E5108: Error executing lua ...l/share/nvim/lazy/tabline.nvim/lua/tabline/highlight.lua:48: attempt to index upvalue
 'NORMAL' (a nil value)
stack traceback:
        ...l/share/nvim/lazy/tabline.nvim/lua/tabline/highlight.lua:48: in function 'get_hl'
        ...l/share/nvim/lazy/tabline.nvim/lua/tabline/highlight.lua:57: in function 'refresh'
        ...l/share/nvim/lazy/tabline.nvim/lua/tabline/highlight.lua:59: in main chunk
        [C]: in function 'require'
        [string ":lua"]:1: in main chunk
```

**Comment by mg979** [comment]
I cannot reproduce. What colorscheme are you using? It seems it fails to get correct values for the `Normal` highlight group.

**Comment by mg979** [comment]
Ok, I could reproduce with default colorscheme and no `termguicolors`.

**Comment by mg979** [comment]
Which is your `termguicolors` setting?

**Comment by wookayin** [comment]
I'm using `termguicolors` in my RC file, but when the plugin is being loaded by plugin manager such as lazy, packer, etc. (esp. for the first time) this setting might not present. So I guess `notermguicolors` were applying when I ran in to this error.

**Comment by mg979** [comment]
Can you try if it works now? Thanks

**Comment by wookayin** [comment]
I think it still happens.

```
  - /tabline.nvim/lua/tabline/setup.lua:159 _in_ **load_theme**
  - /tabline.nvim/lua/tabline/setup.lua:159 _in_ **load_theme**
```

The logic is still buggy; in order to compute `NORMAL = M.get_hl`, the variable `NORMAL` is accessed in Line 60-61.

```
Failed to run `config` for tabline.nvim
.../tabline.nvim/lua/tabline/highlight.lua:61: attempt to index upvalue 'NORMAL' (a nil value)
# stacktrace:
  - /tabline.nvim/lua/tabline/highlight.lua:61 _in_ **get_hl**
  - /tabline.nvim/lua/tabline/highlight.lua:96 _in_ **refresh**
  - /tabline.nvim/lua/tabline/highlight.lua:98
  - .config/nvim/lua/plugins/appearance.lua:20 _in_ **config**
  - .config/nvim/lua/config/plugins.lua:35
  - .dotfiles/nvim/init.lua:49
```

**Comment by mg979** [comment]
Can you try now please? Thanks

**Comment by wookayin** [comment]
Thanks, I think it now works OK -- fixed by 2d891d2.
